### PR TITLE
move singleton to classes's annotations

### PR DESCRIPTION
**Pull request by kunkun-tang**
This patch refactors the Guice uses, and mainly move all singleton binding to respective classes with singleton annotations. The corresponding tests are added as well. a bit more context is at #1285 .

### CODE REVIEW

File: azkaban-web-server/src/test/java/azkaban/webapp/AzkabanWebServerTest.java [hunk 1]
```diff
@@ -118,6 +121,22 @@ public void testInjection() throws Exception {
     final TriggerManager triggerManager1 = requireNonNull(injector.getInstance(TriggerManager.class));
     final TriggerManager triggerManager2 = requireNonNull(injector.getInstance(TriggerManager.class));
     assertTrue(triggerManager1 == triggerManager2);
+
+    //Test if TriggerLoader is singletonly guiced.
+    final TriggerLoader triggerLoader1= requireNonNull(injector.getInstance(TriggerLoader.class));
```
Comment (suvodeep-pyne) [inline azkaban-web-server/src/test/java/azkaban/webapp/AzkabanWebServerTest.java]: Code duplication? Can we create a utility like 
```
assertSingletons(TriggerManager.class, Triggerloader.class, ...);
```

Comment (HappyRay) [inline azkaban-web-server/src/test/java/azkaban/webapp/AzkabanWebServerTest.java]: +1.

To make it useful to test other Guice modules, consider taking an Injector as a parameter.

**Comment by HappyRay** [comment]
Thanks for making this change!

File: azkaban-web-server/src/test/java/azkaban/webapp/AzkabanWebServerTest.java (hunk 1 above)
Comment (kunkun-tang) [inline azkaban-web-server/src/test/java/azkaban/webapp/AzkabanWebServerTest.java]: That's a good suggestion. I am constructing a static assert singleton method in common Test module, such that everybody could call this.

**Comment by kunkun-tang** [comment]
@suvodeep-pyne ExecutorManager code needs to do a save-action refactor before changing guice. Others in azkaban-common module were all added. Web and Exec refactor will be in a separate PR.

**Comment by kunkun-tang** [comment]
I just realized Jamie should be working on ExecutorManager code, so I'm just adding the annotation on top of this class, but would not refactor.

File: azkaban-common/src/test/java/azkaban/ServiceProviderTest.java [hunk 2]
```diff
@@ -39,6 +41,14 @@
 
   public static final String AZKABAN_LOCAL_TEST_STORAGE = "AZKABAN_LOCAL_TEST_STORAGE";
 
+  // Test if one class is singletonly guiced. could be called by
+  // AZ Common, Web, or Exec Modules.
+  public static void assertSingleton(final Class azkabanClass, final Injector injector) {
+    final Object azkabanObj1 = requireNonNull(injector.getInstance(azkabanClass));
```
Comment (HappyRay) [inline azkaban-common/src/test/java/azkaban/ServiceProviderTest.java]: Change requireNonNull to another assertion?

Comment (kunkun-tang) [inline azkaban-common/src/test/java/azkaban/ServiceProviderTest.java]: May I know what is `another assertion` specifically?

Comment (kunkun-tang) [inline azkaban-common/src/test/java/azkaban/ServiceProviderTest.java]: @HappyRay Just updated the code by using AssertJ.

**Comment by kunkun-tang** [comment]
@suvodeep-pyne @HappyRay updated the PR again. Any more comments?

File: azkaban-common/src/test/java/azkaban/ServiceProviderTest.java [hunk 3]
```diff
@@ -44,9 +42,10 @@
   // Test if one class is singletonly guiced. could be called by
   // AZ Common, Web, or Exec Modules.
   public static void assertSingleton(final Class azkabanClass, final Injector injector) {
-    final Object azkabanObj1 = requireNonNull(injector.getInstance(azkabanClass));
-    final Object azkabanObj2 = requireNonNull(injector.getInstance(azkabanClass));
-    assertTrue(azkabanObj1 == azkabanObj2);
+    assertThat(injector.getInstance(azkabanClass)).isNotNull();
+    final Object azkabanObj1 = injector.getInstance(azkabanClass);
+    final Object azkabanObj2 = injector.getInstance(azkabanClass);
+    assertThat(azkabanObj1 == azkabanObj2).isTrue();
```
Comment (HappyRay) [inline azkaban-common/src/test/java/azkaban/ServiceProviderTest.java]: Is http://joel-costigliola.github.io/assertj/core/api/org/assertj/core/api/AbstractAssert.html#isSameAs(java.lang.Object) better?

File: azkaban-common/src/test/java/azkaban/ServiceProviderTest.java [hunk 4]
```diff
@@ -44,9 +42,10 @@
   // Test if one class is singletonly guiced. could be called by
   // AZ Common, Web, or Exec Modules.
   public static void assertSingleton(final Class azkabanClass, final Injector injector) {
-    final Object azkabanObj1 = requireNonNull(injector.getInstance(azkabanClass));
-    final Object azkabanObj2 = requireNonNull(injector.getInstance(azkabanClass));
-    assertTrue(azkabanObj1 == azkabanObj2);
+    assertThat(injector.getInstance(azkabanClass)).isNotNull();
```
Comment (HappyRay) [inline azkaban-common/src/test/java/azkaban/ServiceProviderTest.java]: Why not assert that azkabanObj1 and azkabanObj2 is not null instead?

File: azkaban-common/src/test/java/azkaban/ServiceProviderTest.java (hunk 3 above)
Comment (kunkun-tang) [inline azkaban-common/src/test/java/azkaban/ServiceProviderTest.java]: updated.

File: azkaban-common/src/test/java/azkaban/ServiceProviderTest.java (hunk 4 above)
Comment (kunkun-tang) [inline azkaban-common/src/test/java/azkaban/ServiceProviderTest.java]: This code is outdated in commt1.

File: azkaban-common/src/test/java/azkaban/ServiceProviderTest.java [hunk 5]
```diff
@@ -39,6 +39,15 @@
 
   public static final String AZKABAN_LOCAL_TEST_STORAGE = "AZKABAN_LOCAL_TEST_STORAGE";
 
+  // Test if one class is singletonly guiced. could be called by
+  // AZ Common, Web, or Exec Modules.
+  public static void assertSingleton(final Class azkabanClass, final Injector injector) {
+    assertThat(injector.getInstance(azkabanClass)).isNotNull();
+    final Object azkabanObj1 = injector.getInstance(azkabanClass);
+    final Object azkabanObj2 = injector.getInstance(azkabanClass);
+    assertThat(azkabanObj1).isEqualTo(azkabanObj2);
```
Comment (HappyRay) [inline azkaban-common/src/test/java/azkaban/ServiceProviderTest.java]: Shouldn't it be http://joel-costigliola.github.io/assertj/core/api/org/assertj/core/api/AbstractAssert.html#isSameAs(java.lang.Object) 
isSameAs?

File: azkaban-common/src/test/java/azkaban/ServiceProviderTest.java (hunk 4 above)
Comment (HappyRay) [inline azkaban-common/src/test/java/azkaban/ServiceProviderTest.java]: I don't see this comment addressed in the latest code.

**Pull request merged**